packages/@react-aria/interactions/src/usePress.ts

```diff
@@ -228,27 +228,48 @@ export function usePress(props: PressHookProps): PressResult {
 
     let pressProps: HTMLAttributes<HTMLElement> = {
       onKeyDown(e) {
-        if (isValidKeyboardEvent(e.nativeEvent) && e.currentTarget.contains(e.target as HTMLElement)) {
-          e.preventDefault();
-          e.stopPropagation();
-
-          // If the event is repeating, it may have started on a different element
-          // after which focus moved to the current element. Ignore these events and
-          // only handle the first key down event.
-          if (!state.isPressed && !e.repeat) {
-            state.target = e.currentTarget as HTMLElement;
-            state.isPressed = true;
-            triggerPressStart(e, 'keyboard');
-
-            // Focus may move before the key up event, so register the event on the document
-            // instead of the same element where the key down event occurred.
-            addGlobalListener(document, 'keyup', onKeyUp, false);
+        const {currentTarget, target, nativeEvent, repeat} = e;
+        if (currentTarget.contains(target as HTMLElement)) {
+          const isValid = isValidKeyboardEvent(nativeEvent);
+          if (isValid) {
+            e.stopPropagation();
+          }
+          if (isValid ||
+            (
+              currentTarget !== target &&
+              (isHTMLAnchorLink(target as HTMLElement) || (target as HTMLElement).getAttribute('role') === 'link') &&
+              isSpaceKey(nativeEvent)
+            )
+          ) {
+            e.preventDefault();
+            // If the event is repeating, it may have started on a different element
+            // after which focus moved to the current element. Ignore these events and
+            // only handle the first key down event.
+            if (!state.isPressed && !repeat) {
+              state.target = currentTarget as HTMLElement;
+              state.isPressed = true;
+              triggerPressStart(e, 'keyboard');
+
+              // Focus may move before the key up event, so register the event on the document
+              // instead of the same element where the key down event occurred.
+              addGlobalListener(document, 'keyup', onKeyUp, false);
+            }
           }
         }
       },
       onKeyUp(e) {
-        if (isValidKeyboardEvent(e.nativeEvent) && !e.repeat && e.currentTarget.contains(e.target as HTMLElement)) {
-          triggerPressUp(createEvent(state.target, e), 'keyboard');
+        const {currentTarget, target, nativeEvent, repeat} = e;
+        if (!repeat && currentTarget.contains(target as HTMLElement)) {
+          if (
+            isValidKeyboardEvent(nativeEvent) ||
+            (
+              currentTarget !== target &&
+              (isHTMLAnchorLink(target as HTMLElement) || (target as HTMLElement).getAttribute('role') === 'link') &&
+              isSpaceKey(nativeEvent)
```

Comment:
This conditional is the same as above and makes both hard to read. Can we make it a helper method for clarity?